rt/rs/security/http-signature/src/main/java/org/apache/cxf/rs/security/httpsignature/MessageVerifier.java

```diff
@@ -0,0 +1,212 @@
+package org.apache.cxf.rs.security.httpsignature;
+
+import org.apache.cxf.common.logging.LogUtils;
+import org.tomitribe.auth.signatures.Signature;
+import org.tomitribe.auth.signatures.Verifier;
+
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.Security;
+import java.util.*;
+import java.util.logging.Logger;
+
+public class MessageVerifier {
+    private ExceptionHandler exceptionHandler;
+
+    private static final Logger LOG = LogUtils.getL7dLogger(MessageVerifier.class);
+
+    public MessageVerifier(boolean verifyMessageBody) {
+        setExceptionHandler(null);
+        setSecurityProvider(null);
+        setAlgorithmProvider(null);
+        this.verifyMessageBody = verifyMessageBody;
+    }
+
+    public MessageVerifier(PublicKeyProvider publicKeyProvider,
+                           ExceptionHandler exceptionHandler,
+                           SecurityProvider securityProvider,
+                           AlgorithmProvider algorithmProvider,
+                           boolean verifyMessageBody)
+    {
+        Objects.requireNonNull(publicKeyProvider, "Public key provider cannot be null");
+        this.publicKeyProvider = publicKeyProvider;
+        this.exceptionHandler = exceptionHandler;
+        this.securityProvider = securityProvider;
+        this.algorithmProvider = algorithmProvider;
+        this.verifyMessageBody = verifyMessageBody;
+    }
+
+    private PublicKeyProvider publicKeyProvider;
+
+    private SecurityProvider securityProvider;
+
+    private AlgorithmProvider algorithmProvider;
+
+    private boolean verifyMessageBody;
+
+    public void setSecurityProvider(SecurityProvider securityProvider) {
+        if (securityProvider != null) {
+            this.securityProvider = securityProvider;
+        } else {
+            this.securityProvider = (keyId) -> Security.getProvider("SunRsaSign");
```

Comment:
Should me make it default and use `Objects.requireNonNull`? The logic of the setter is a bit unexpected